From 87a1cd88660aac4c3d6d1ae46efe7fe1c1e65d80 Mon Sep 17 00:00:00 2001
From: Han Gao <gaohan@iscas.ac.cn>
Date: Wed, 20 May 2026 00:55:46 +0800
Subject: [PATCH 165/440] FROMLIST: riscv: unconditionally select
 ARCH_KEEP_MEMBLOCK

Select ARCH_KEEP_MEMBLOCK unconditionally. kexec requires memblock
to be kept after boot to initialize the secondary kernel. Device
Tree platforms also need this for kexec support.

Link: https://lore.kernel.org/r/20260519165546.123105-1-gaohan@iscas.ac.cn
Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
---
 arch/riscv/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/riscv/Kconfig b/arch/riscv/Kconfig
index d75c95ed611ec..eea01ab744b99 100644
--- a/arch/riscv/Kconfig
+++ b/arch/riscv/Kconfig
@@ -55,7 +55,7 @@ config RISCV
 	select ARCH_HAS_UBSAN
 	select ARCH_HAS_VDSO_ARCH_DATA if HAVE_GENERIC_VDSO
 	select ARCH_HAVE_NMI_SAFE_CMPXCHG
-	select ARCH_KEEP_MEMBLOCK if ACPI
+	select ARCH_KEEP_MEMBLOCK
 	select ARCH_MHP_MEMMAP_ON_MEMORY_ENABLE	if 64BIT && MMU
 	select ARCH_OPTIONAL_KERNEL_RWX if ARCH_HAS_STRICT_KERNEL_RWX
 	select ARCH_OPTIONAL_KERNEL_RWX_DEFAULT
-- 
2.55.0

